### PR TITLE
Add mps support.

### PR DESCRIPTION
**Pull request by iamnotagentleman**
## Describe your changes and approach used

Added MPS support for better performance on apple silicon.
Added __init__.py in models/ to make it a Python package.

## TEST MACHINE
``` 

{'cpu_count': 11,
 'machine': 'arm64',
 'platform': 'macOS-14.4.1-arm64-arm-64bit',
 'processor': 'arm',
 'ram': '36 GB',
 'system': 'Darwin'}

```

### CODE REVIEW

File: lang_sam/models/gdino.py [hunk 1]
```diff
@@ -18,7 +21,9 @@ def __init__(self):
     def build_model(self, ckpt_path: str | None = None):
         model_id = "IDEA-Research/grounding-dino-base"
         self.processor = AutoProcessor.from_pretrained(model_id)
-        self.model = AutoModelForZeroShotObjectDetection.from_pretrained(model_id).to(DEVICE)
+        self.model = AutoModelForZeroShotObjectDetection.from_pretrained(model_id).to(
```
Comment (iamnotagentleman) [inline lang_sam/models/gdino.py:24]: (linter)

**Comment by luca-medeiros** [comment]
Cool! guess we can do the same here https://github.com/luca-medeiros/lang-segment-anything/blob/d3347fa3d6117fa9e2d56d445e2d3db4e48cd9f8/lang_sam/models/sam.py#L9

**Comment by iamnotagentleman** [comment]
Great catch, @luca-medeiros! I’ve also added some additional information regarding this PR.


MPS users will encounter warnings like the following:

`UserWarning: The operator 'aten::upsample_bicubic2d.out' is not currently supported on the MPS backend and will fall back to run on the CPU. This may have performance implications. (Triggered internally at /Users/runner/work/pytorch/pytorch/pytorch/aten/src/ATen/mps/MPSFallback.mm:13.)
`
As of version 2.4.1, upsample_bicubic2d.out is not supported on the MPS backend. However, support has been added to the nightly builds, so native support of this function should be available soon.

Additionally, autocast support for MPS is on the way, though there is no planned merge yet. Once available, it should improve performance even further.

**Comment by luca-medeiros** [comment]
@iamnotagentleman Nice, didn't know about that. Appreciate the explanation and PR!
What do you think would be best? wait for the stable release or revisit later?

**Comment by iamnotagentleman** [comment]
@luca-medeiros, I believe we can proceed for now. Other operations besides "upsample_bicubic2d" are still accelerated by MPS, and SAM-2 has received optimizations for the MPS infrastructure, so overall performance should still improve. I haven’t encountered any bugs so far.

**Pull request merged**